meta-mender-core/recipes-mender/mender-client/mender-client.inc

```diff
@@ -373,23 +192,84 @@ do_install:append:class-target:mender-growfs-data:mender-systemd() {
 }
 
 do_install:append:class-target:mender-persist-systemd-machine-id() {
-    install -m 644 ${WORKDIR}/mender-client-systemd-machine-id.service ${D}${systemd_unitdir}/system/
+    install -m 644 ${WORKDIR}/mender-systemd-machine-id.service ${D}${systemd_unitdir}/system/
     install -d -m 755 ${D}${systemd_unitdir}/system/${MENDER_CLIENT}.service.wants
-    ln -sf ../mender-client-systemd-machine-id.service ${D}${systemd_unitdir}/system/${MENDER_CLIENT}.service.wants/
+    ln -sf ../mender-systemd-machine-id.service ${D}${systemd_unitdir}/system/${MENDER_CLIENT}.service.wants/
     install -d -m 755 ${D}${bindir}
-    install -m 755 ${WORKDIR}/mender-client-set-systemd-machine-id.sh ${D}${bindir}/
+    install -m 755 ${WORKDIR}/mender-set-systemd-machine-id.sh ${D}${bindir}/
 }
 
-# Native overrides, installing scripts for package mender-modules-gen
-do_compile:class-native() {
-    true
-}
-
-do_install:class-native() {
-    oe_runmake \
-        -C ${B}/src/${GO_IMPORT} \
-        V=1 \
-        prefix=${D} \
-        bindir=${bindir} \
-        install-modules-gen
-}
+def mender_maybe_artifact_info(d):
+    import re
+    branch = d.getVar("MENDER_BRANCH")
+    match = re.match(r"([0-9]+)\.([0-9]+)\.", branch)
+    if match is None:
+        return ""
+
+    major = int(match.group(1))
+    minor = int(match.group(2))
+    if major < 3 or (major == 3 and minor < 5):
+        return " mender-artifact-info"
+    else:
+        return ""
+
+# The revision listed below is not really important, it's just a way to avoid
+# network probing during parsing if we are not gonna build the git version
+# anyway. If git version is enabled, the AUTOREV will be chosen instead of the
+# SHA.
+def mender_autorev_if_git_version(d):
+    version = d.getVar("PREFERRED_VERSION")
+    if version is None or version == "":
+        version = d.getVar("PREFERRED_VERSION_%s" % d.getVar('PN'))
+    if not d.getVar("EXTERNALSRC") and version is not None and "git" in version:
+        return d.getVar("AUTOREV")
+    else:
+        return "f6ffa190892202263fdb75975059fbb201adab6a"
+
+def mender_branch_from_preferred_version(d):
+    import re
+    version = d.getVar("PREFERRED_VERSION")
+    if version is None or version == "":
+        version = d.getVar("PREFERRED_VERSION_%s" % d.getVar('PN'))
+    if version is None:
+        version = ""
+    match = re.match(r"^[0-9]+\.[0-9]+\.", version)
+    if match is not None:
+        # If the preferred version is some kind of version, use the branch name
+        # for that one (1.0.x style).
+        return match.group(0) + "x"
+    else:
+        # Else return master as branch.
+        return "master"
+
+def mender_version_from_preferred_version(d):
```

Comment:
We shouldn't keep these in a file which is shared between official recipes and Git recipes. There is some hacky stuff in the Git recipes, and we don't want that to "leak" into the official recipes (for example the license check is less strict).